File: Backend/src/main/java/lan/dk/podcastserver/business/find/FindPodcastBusiness.java
package lan.dk.podcastserver.business.find;

import lan.dk.podcastserver.entity.Podcast;
import lan.dk.podcastserver.manager.worker.selector.FinderSelector;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

/**
 * Created by kevin on 22/02/15 for Podcast Server
 */
@Component
@RequiredArgsConstructor
public class FindPodcastBusiness {

    private final FinderSelector finderSelector;
    
    public Podcast fetchPodcastInfoByUrl(String url) {
        return finderSelector.of(url).find(url);
    }
}

File: backend/src/main/kotlin/com/github/davinkevin/podcastserver/business/find/FindPodcastBusiness.kt
package com.github.davinkevin.podcastserver.business.find

import lan.dk.podcastserver.entity.Podcast
import com.github.davinkevin.podcastserver.manager.selector.FinderSelector
import org.springframework.stereotype.Component

/**
 * Created by kevin on 22/02/15 for Podcast Server
 */
@Component
class FindPodcastBusiness(val finderSelector: FinderSelector) {
    fun fetchPodcastInfoByUrl(url: String): Podcast = finderSelector.of(url).find(url)
}
